### DISCUSSION TITLE
Best way to debug compile error.

### DISCUSSION THREAD

**Discussion by zouhangwithsweet** [web-infra-dev/rsbuild#684] [General]
In my project. There are some errors like:

<img width="1005" alt="image" src="https://github.com/web-infra-dev/rsbuild/assets/21095710/b19d1104-294e-4e31-9110-46a54621e277">

I think the `path` module should be resolve, beaucase it's builtin module. How to debug this kind error. Is there some best practices? :)

**Comment by chenjiahan** [comment]
How is this babel plugin been used? Can You provide the rsbuild.config.ts content

**Reply by zouhangwithsweet** [reply]
Nothing in `rsbuild.config.ts` expect `react-plugin`.  It's the same craco project in our last discussion.

**Reply by chenjiahan** [reply]
So babel-plugin-marco is imported in the client code? Maybe you need https://rsbuild.dev/plugins/list/plugin-node-polyfill

**Reply by zouhangwithsweet** [reply]
Wow, I will test it.

**Reply by madwau** [reply]
I came across the same problem and wasn't sure either how to debug this. The plugin helped.

![OnPaste 20231203-162857](https://github.com/web-infra-dev/rsbuild/assets/7109225/0f8777b9-c2db-4a87-8bcf-522ea3a10cf8)

**Comment by chenjiahan** [comment]
For simple errors, we just use the Rsbuild debug mode to inspect the Rspack config: https://rsbuild.dev/guide/debug/debug-mode.html

If the issue is more complex, we will provide Rsbuild Doctor as a debug tool in the future. At this stage, you may need to debug the source code of Rspack
